cumulusci/core/config/OrgConfig.py

```diff
@@ -489,27 +489,35 @@ def is_advanced_currency_management_enabled(self):
             return False
 
     def resolve_04t_dependencies(self, dependencies):
-        """Look up 04t SubscriberPackageVersion ids for 1gp project dependencies"""
+        """Look up 04t SubscriberPackageVersion ids for 1GP project dependencies"""
+        from cumulusci.core.dependencies.dependencies import (
+            PackageNamespaceVersionDependency,
+            PackageVersionIdDependency,
+        )
+
+        # Circular dependency.
+
         new_dependencies = []
         for dependency in dependencies:
-            dependency = {**dependency}
-
-            if "namespace" in dependency:
+            if (
+                isinstance(dependency, PackageNamespaceVersionDependency)
+                and dependency.namespace
```

Comment:
If it's a PackageNamespaceVersionDependency then we can assume it has a namespace, right?

Reply:
Yep, good catch. Fixed.